examples/server/server.cpp

```diff
@@ -1162,6 +1164,16 @@ struct server_metrics {
     }
 };
 
+struct termination_signal {
+    int number;
+};
+
+struct standby_timeout {};
+
+using shutdown_reason = std::variant<termination_signal, standby_timeout>;
+
+std::function<void(shutdown_reason)> shutdown_handler;
```

Comment:
we already had `std::function<void(int)> shutdown_handler`, why we ever need to create a new one?